db/sql/migrations/updates/20180829161959_remove_duplicate_rows_and_add_primary_key_in_dapps.sql

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+BEGIN;
+	DELETE FROM "dapps" "d0"
+		USING "dapps" "d1"
+	WHERE "d0"."ctid" < "d1"."ctid"
+		AND "d0"."transactionId" = "d1"."transactionId";
+
+	ALTER TABLE "dapps"
+		ADD PRIMARY KEY ("transactionId");
```

Comment:
I will never prefer to have foreign key to be a primary key as well. It's against database ACID property of database. 

Yes people have mixed opinion on the topic in general (not related to PostgreSQL), you can look for it as well.
https://www.quora.com/Can-I-have-the-same-field-as-Primary-Key-and-Foreign-Key-in-MySQL
https://stackoverflow.com/questions/10982992/is-it-fine-to-have-foreign-key-as-primary-key


So prefer to use stuff which have common ground. The actual problem was we had an index on that column 

https://github.com/LiskHQ/lisk/blob/8f7de3963eaa48a6e2d8b081de37921b85911a53/db/sql/migrations/updates/20160723182900_create_schema.sql#L167
We just need to make that index unique, and keep the foreign key constraint as it is.

Reply:
I agree with what you said. My thought processes was to ensure consistency, because `transfer` and `signatures` table had `transactionId` as primary key. But looking further, i found that other asset tables like `delegates` and `outtransfer` and `intransfer` use unique index. I'll make the index unique.

Reply:
I would prefer to use a unique constraint.
https://www.postgresql.org/docs/9.6/static/ddl-constraints.html

`
Note: The preferred way to add a unique constraint to a table is ALTER TABLE ... ADD CONSTRAINT. The use of indexes to enforce unique constraints could be considered an implementation detail that should not be accessed directly. One should, however, be aware that there's no need to manually create indexes on unique columns; doing so would just duplicate the automatically-created index.
`